### DISCUSSION TITLE
Auto-filtering of entity drop down list  - Energy dashboard configuration

### DISCUSSION THREAD

**Discussion by chadbaldwin** [home-assistant/frontend#15291] [Other feature requests]
I noticed recently that when you are using the History page that as you add entities they are automatically removed from the list of options in the "Choose entity" dropdown. This makes life so much easier when you are trying to add many devices. For example, I have 20+ devices named "outlet_##_power". Simply being able to type "_power" into a dropdown, and have it filter down to ONLY the entities which match that criteria and have not yet been added makes it quick to complete the task.

While obviously this would be nice to have in ALL places where there is any sort of "Choose entity" drop down box and it is in a place where it would make sense to have auto filtering....I figure that might be too broad of a feature request to ask for and maybe it would be easier to request these items as piecemeal.

So this feature request dicussion is specifically in regard to adding this same filtering capability to the various "Add {category}" menus when configuring the Energy Dashboard.

It would make adding new/missing entities to track significantly easier.

**Comment by ildar170975** [comment]
In some cases filtering entities may be WRONG.
Imagine an Entities card: if you do not allow to select some entity twice - then what will happen? ))

**Reply by chadbaldwin** [reply]
I agree, that's why I said "and it is in a place where it would make sense to have auto filtering" 😉

**Reply by ildar170975** [reply]
Probably only here:
- automations via UI (could be already done... cannot check myself);
- dev tools -> call a service (seems to be already done);
- Logbook page (when it becomes looking similarly to History page - i.e. where several entities may be selected manually, currently only one entity may be manually selected).

**Reply by chadbaldwin** [reply]
And also on all of the `Add {category}` sections of the energy dashboard, which is what this discussion was specifically targeted toward since requesting it across the board would be too vague and each instance would need to be handled and solved differently.
